prime-router/src/main/kotlin/serializers/Hl7Serializer.kt

```diff
@@ -233,7 +233,7 @@ class Hl7Serializer(
         val errors = mutableListOf<String>()
         val warnings = mutableListOf<String>()
         // key of the map is the column header, list is the values in the column
-        val mappedRows: MutableMap<String, MutableSet<String>> = mutableMapOf()
+        val mappedRows: MutableMap<String, String> = mutableMapOf()
```

Comment:
It seems wrong to me that you have to make changes to both Hl7Serializer and CsvSerializer.

Again, in the heavenly @whytheplatypus Theory world, it seems like we'd transform to FHIR as early as computerly possible, then do all transforms in the FHIR world.   Deserializers and Serializers would be blissfully unaware of such things as mappers/transforms.    True?

Reply:
Yeah, the problem is how the two serializers run through the same process with different logic.